airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/api.py

```diff
@@ -104,17 +114,19 @@ def handle_call_rate_limit(self, response, params):
                 max_usage = max(max_usage, usage)
                 max_pause_interval = max(max_pause_interval, pause_interval)
 
-            if max_usage > self.call_rate_threshold:
-                max_pause_interval = max(max_pause_interval, self.pause_interval_minimum)
-                logger.warning(f"Utilization is too high ({max_usage})%, pausing for {max_pause_interval}")
-                sleep(max_pause_interval.total_seconds())
+            if max_usage >= self.MIN_RATE:
+                sleep_time = self.compute_pause_interval(usage=max_usage, pause_interval=max_pause_interval)
+                logger.warning(f"Utilization is too high ({max_usage})%, pausing for {sleep_time}")
+                sleep(sleep_time).total_seconds()
+
+
         else:
             headers = response.headers()
             usage, pause_interval = self._parse_call_rate_header(headers)
-            if usage > self.call_rate_threshold or pause_interval:
-                pause_interval = max(pause_interval, self.pause_interval_minimum)
-                logger.warning(f"Utilization is too high ({usage})%, pausing for {pause_interval}")
-                sleep(pause_interval.total_seconds())
+            if usage >= self.MIN_RATE or pause_interval:
```

Comment:
What do you think of computing the usage and pause interval for "batch" in a separate function, so that the output of "batch" and "not batch" share the same rate-limiting logic (which currently looks duplicated).